packages/nimbus/src/components/date-picker/date-picker.stories.tsx

```diff
@@ -0,0 +1,2509 @@
+        // Invisible
+        await expect(segments[1]).toHaveAttribute("aria-valuenow", "15");
+        await userEvent.keyboard("{ArrowDown}");
+        // Now visible
+        await expect(segments[1]).toHaveAttribute("aria-valuenow", "15");
+
+        // Move to year and modify
+        await userEvent.tab(); // Move to year segment
+        // Invisible
+        await expect(segments[2]).toHaveAttribute("aria-valuenow", "2025");
+        await userEvent.keyboard("{ArrowUp}");
+        // Now visible
+        await expect(segments[2]).toHaveAttribute("aria-valuenow", "2025");
+
+        // Now that we have a complete date, clear button should be enabled
+        const clearButton = within(withPlaceholder).getByRole("button", {
+          name: /clear/i,
+        });
+        await expect(clearButton).not.toBeDisabled();
+      }
+    );
+
+    await step(
+      "Placeholder value doesn't affect typing input directly",
+      async () => {
+        // Clear the first DatePicker to reset it
+        const withPlaceholder = canvas.getByRole("group", {
+          name: "Date picker with placeholder",
+        });
+        const clearButton = within(withPlaceholder).getByRole("button", {
+          name: /clear/i,
+        });
+
+        // Only click clear if it's enabled
+        if (!clearButton.hasAttribute("disabled")) {
+          await userEvent.click(clearButton);
+        }
+
+        const segments = within(withPlaceholder).getAllByRole("spinbutton");
+
+        // Type directly into segments - this should work independently of placeholder value
+        await userEvent.click(segments[0]); // month
+        await userEvent.keyboard("12"); // December
+
+        await userEvent.click(segments[1]); // day
+        await userEvent.keyboard("25"); // 25th
+
+        await userEvent.click(segments[2]); // year
+        await userEvent.keyboard("2024"); // 2024
+
+        // Values should be exactly what was typed, not influenced by placeholder
+        await expect(segments[0]).toHaveAttribute("aria-valuenow", "12");
+        await expect(segments[1]).toHaveAttribute("aria-valuenow", "25");
+        await expect(segments[2]).toHaveAttribute("aria-valuenow", "2024");
+
+        // Clear button should be enabled since we have a value
+        const clearButtonAfterTyping = within(withPlaceholder).getByRole(
+          "button",
+          {
+            name: /clear/i,
+          }
+        );
+        await expect(clearButtonAfterTyping).not.toBeDisabled();
+      }
+    );
+  },
+};
+
+/**
+ * Variants, Sizes, and States
+ * Demonstrates all combinations of available sizes, variants, and form states
+ */
+export const VariantsSizesAndStates: Story = {
+  args: {
+    ["aria-label"]: "Select a date",
+  },
+  render: (args) => {
+    const states = [
+      { label: "Default", props: {} },
+      { label: "Disabled", props: { isDisabled: true } },
+      { label: "Read Only", props: { isReadOnly: true } },
```

Comment:
The clear button is not `disabled` and can still update the value of the picker when `isDisabled` or `isReadOnly` are `true`.